sonar-scanner does not start properly: java is lacking tzdb.dat

This issue was hidden by #719 

I am seeing the following on startup of locally testing the go test task:
```
 go test -v -count=1 -timeout ${ODS_TESTTIMEOUT:-30m} ./test/tasks/... --run 'TestTaskODSBuildGo/build_go_app$'
=== RUN   TestTaskODSBuildGo
...
=== RUN   TestTaskODSBuildGo/build_go_app
    run.go:141: Workspace is in /Users/henrichkraemer_1/code/bix/ods2/ods-pipeline/test/testdata/workspaces/workspace-781435638
2023/09/01 10:57:05 Waiting up to 300 seconds for task ods-build-go-taskrun-tngmlgum in namespace jypvivtn to be done...
2023/09/01 10:57:05 Waiting for pod related to TaskRun ods-build-go-taskrun-tngmlgum to be added to the cluster
2023/09/01 10:57:05 TaskRun ods-build-go-taskrun-tngmlgum added pod ods-build-go-taskrun-tngmlgum-pod to the cluster
2023/09/01 10:57:05 Waiting for container step-build-go-binary from pod ods-build-go-taskrun-tngmlgum-pod to be ready...
2023/09/01 10:57:05 Watching events for pod ods-build-go-taskrun-tngmlgum-pod in namespace jypvivtn
2023/09/01 10:57:05 ---------------------- Events -------------------------
...
2023/09/01 10:57:10 Type: Normal, Message: Successfully pulled image "localhost:5000/ods/ods-sonar:latest" in 274.509791ms
2023/09/01 10:57:10 Type: Normal, Message: Created container step-scan-with-sonar
2023/09/01 10:57:10 Type: Normal, Message: Started container step-scan-with-sonar
2023/09/01 10:57:11 ---------------------- Logs from step-build-go-binary -------------------------
...
2023/09/01 10:57:23 Waiting for container step-scan-with-sonar from pod ods-build-go-taskrun-tngmlgum-pod to be ready...
2023/09/01 10:57:23 ---------------------- Logs from step-scan-with-sonar -------------------------
INFO  | Scanning with sonar-scanner ...
INFO: Scanner configuration file: /usr/local/sonar-scanner-cli/conf/sonar-scanner.properties
INFO: Project root configuration file: /workspace/source/sonar-project.properties
INFO: SonarScanner 4.6.2.2472
INFO: Java 11.0.20 Red Hat, Inc. (64-bit)
INFO: Linux 5.15.49-linuxkit-pr aarch64
INFO: SONAR_SCANNER_OPTS=-Djavax.net.ssl.trustStore=/usr/lib/jvm/jre-11/lib/security/cacerts -Djavax.net.ssl.trustStorePassword=changeit
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 0.025s
INFO: Final Memory: 1M/17M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarScanner execution
java.lang.Error: java.io.FileNotFoundException: /usr/lib/jvm/java-11-openjdk-11.0.20.0.8-2.el8.aarch64/lib/tzdb.dat (No such file or directory)
	at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:261)
	at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:251)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.util.calendar.ZoneInfoFile.<clinit>(ZoneInfoFile.java:251)
	at java.base/sun.util.calendar.ZoneInfo.getTimeZone(ZoneInfo.java:588)
	at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:577)
	at java.base/java.util.TimeZone.getTimeZone(TimeZone.java:517)
	at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.Util.<clinit>(Util.java:85)
	at org.sonarsource.scanner.api.internal.shaded.okhttp.OkHttpClient.<clinit>(OkHttpClient.java:124)
	at org.sonarsource.scanner.api.internal.shaded.okhttp.OkHttpClient$Builder.<init>(OkHttpClient.java:449)
	at org.sonarsource.scanner.api.internal.OkHttpClientFactory.create(OkHttpClientFactory.java:73)
	at org.sonarsource.scanner.api.internal.ServerConnection.<init>(ServerConnection.java:52)
	at org.sonarsource.scanner.api.internal.ServerConnection.create(ServerConnection.java:62)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.createLauncher(IsolatedLauncherFactory.java:67)
	at org.sonarsource.scanner.api.EmbeddedScanner.doStart(EmbeddedScanner.java:185)
	at org.sonarsource.scanner.api.EmbeddedScanner.start(EmbeddedScanner.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:73)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.io.FileNotFoundException: /usr/lib/jvm/java-11-openjdk-11.0.20.0.8-2.el8.aarch64/lib/tzdb.dat (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/sun.util.calendar.ZoneInfoFile$1.run(ZoneInfoFile.java:255)
	... 17 more

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ listed in the changelog.
 
 - update sonar-scanner and cnes-report ([#725](https://github.com/opendevstack/ods-pipeline/issues/725))
 
+- SonarQube doesn't scan FE-related code ([#716](https://github.com/opendevstack/ods-pipeline/issues/716))
 
 ## [0.13.2] - 2023-07-18
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,9 @@ listed in the changelog.
 
 - sonar-scanner does not start properly: java is lacking tzdb.dat ([#723](https://github.com/opendevstack/ods-pipeline/issues/723))
 
+- update sonar-scanner and cnes-report ([#725](https://github.com/opendevstack/ods-pipeline/issues/725))
+
+
 ## [0.13.2] - 2023-07-18
 
 ### Fixed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ listed in the changelog.
 
 - sonar-scanner invocations stderr not captured ([#719](https://github.com/opendevstack/ods-pipeline/issues/719))
 
+- sonar-scanner does not start properly: java is lacking tzdb.dat ([#723](https://github.com/opendevstack/ods-pipeline/issues/723))
+
 ## [0.13.2] - 2023-07-18
 
 ### Fixed

--- a/build/package/Dockerfile.sonar
+++ b/build/package/Dockerfile.sonar
@@ -4,8 +4,8 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 USER root
 WORKDIR /usr/src/app
 
-ENV SONAR_SCANNER_VERSION=4.6.2.2472 \
-    CNES_REPORT_VERSION=4.0.0
+ENV SONAR_SCANNER_VERSION=4.8.0.2856 \
+    CNES_REPORT_VERSION=4.2.0
 
 # Build Go binary.
 COPY go.mod .
@@ -37,7 +37,7 @@ ENV SONAR_EDITION="community" \
     JAVA_HOME=/usr/lib/jvm/jre-11
 
 RUN microdnf install --nodocs java-11-openjdk-headless tzdata-java which && microdnf clean all
- # tzdata-java was missing unless explicitly installed: see https://stackoverflow.com/a/76848186
+# tzdata-java was missing unless explicitly installed: see https://stackoverflow.com/a/76848186
 
 COPY --from=builder /usr/local/bin/sonar /usr/local/bin/sonar
 COPY --from=builder /usr/local/sonar-scanner-cli /usr/local/sonar-scanner-cli

--- a/build/package/Dockerfile.sonar
+++ b/build/package/Dockerfile.sonar
@@ -31,13 +31,25 @@ RUN cd /tmp \
     && chmod +x /usr/local/cnes/cnesreport.jar
 
 # Final image
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4
+FROM registry.access.redhat.com/ubi8/nodejs-18:1
+
+ENV NPM_CONFIG_PREFIX=$HOME/.npm-global \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8
+
+RUN echo id: $(id) && \
+    echo node version: $(node --version) && \
+    echo npm version: $(npm --version) && \
+    echo npx version: $(npx --version)
 
 ENV SONAR_EDITION="community" \
     JAVA_HOME=/usr/lib/jvm/jre-11
 
-RUN microdnf install --nodocs java-11-openjdk-headless tzdata-java which && microdnf clean all
-# tzdata-java was missing unless explicitly installed: see https://stackoverflow.com/a/76848186
+USER root
+RUN INSTALL_PKGS="java-11-openjdk-headless which" && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum -y clean all --enablerepo='*'
 
 COPY --from=builder /usr/local/bin/sonar /usr/local/bin/sonar
 COPY --from=builder /usr/local/sonar-scanner-cli /usr/local/sonar-scanner-cli

--- a/build/package/Dockerfile.sonar
+++ b/build/package/Dockerfile.sonar
@@ -36,7 +36,8 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4
 ENV SONAR_EDITION="community" \
     JAVA_HOME=/usr/lib/jvm/jre-11
 
-RUN microdnf install --nodocs java-11-openjdk-headless which && microdnf clean all
+RUN microdnf install --nodocs java-11-openjdk-headless tzdata-java which && microdnf clean all
+ # tzdata-java was missing unless explicitly installed: see https://stackoverflow.com/a/76848186
 
 COPY --from=builder /usr/local/bin/sonar /usr/local/bin/sonar
 COPY --from=builder /usr/local/sonar-scanner-cli /usr/local/sonar-scanner-cli

--- a/docs/design/software-design-specification.adoc
+++ b/docs/design/software-design-specification.adoc
@@ -557,13 +557,13 @@ a| The script installs the Helm chart located in `deploy/ods-pipeline`. Further,
 
 | SDS-EXT-7
 | sonar-scanner
-| 4.6
+| 4.8
 | General purpose SonarQube scanner
 | https://github.com/SonarSource/sonar-scanner-cli
 
 | SDS-EXT-8
 | cnes-report
-| 4.0
+| 4.2
 | Exports code analysis from a SonarQube server in various file formats.
 | https://github.com/cnescatlab/sonar-cnes-report
 

--- a/docs/design/software-design-specification.adoc
+++ b/docs/design/software-design-specification.adoc
@@ -28,7 +28,7 @@ As described in the architecture, the system is installed into local namespaces.
 |===
 | SDS-SHARED-1
 | `ods-sonar` container image
-| Container image for SQ scanning. Based on `ubi8/ubi-minimal` (SDS-EXT-2), includes software to analyze source code statically (SDS-SHARED-2, SDS-EXT-7, SDS-EXT-8 and SDS-EXT-30).
+| Container image for SQ scanning. Based on `ubi8/nodejs-18` (SDS-EXT-32), includes software to analyze source code statically (SDS-SHARED-2, SDS-EXT-7, SDS-EXT-8 and SDS-EXT-30).
 
 | SDS-SHARED-2
 | `sonar` binary
@@ -380,7 +380,7 @@ If the Aqua scanner is installed in the base image, the pushed image shall be sc
 
 Processes tags specified in the `extra-tags` parameter to add tags to the image in the target registry:
 
-* If an artifact for the tagged image exists, the tag is not further processed. 
+* If an artifact for the tagged image exists, the tag is not further processed.
 * If there is no artifact for the tagged image:
     * Adds the tag using skopeo (SDS-EXT-17) to the image in the target registry.
     * The resulting tagged image artifact is placed into `.ods/artifacts`.

